lib/http/timeout/null.rb

```diff
@@ -46,6 +46,28 @@ def write(data)
         @socket << data
       end
 
+      # Retry reading
+      def __rescue_readable(&block)
```

Comment:
Yes, capturing a block as an object allocates memory and is therefore slower. Not sure if it matters here...